src/veh_interact.cpp

```diff
@@ -1349,20 +1318,13 @@ int veh_interact::part_at (int dx, int dy)
 }
 
 /**
- * Checks to see if you can currently install this part at current position.
+ * Checks to see if you can potentially install this part at current position.
  * Affects coloring in display_list() and is also used to
- * sort can_mount so installable parts come first.
+ * sort can_mount so potentially installable parts come first.
  */
-bool veh_interact::can_currently_install(const vpart_info &vpart)
+bool veh_interact::can_potentially_install(const vpart_info &vpart)
 {
-    if (g->u.has_trait("DEBUG_HS")) {
-        return true;
-    }
-    bool has_comps = crafting_inv.has_components(vpart.item, 1);
-    ///\EFFECT_MECHANICS determines which vehicle parts can be installed
-    bool has_skill = g->u.get_skill_level( skill_mechanics ) >= vpart.difficulty;
-    bool is_wheel = vpart.has_flag("WHEEL");
-    return (has_comps && (has_skill || is_wheel));
+    return g->u.has_trait( "DEBUG_HS" ) || vpart.install_reqs.can_make_with_inventory( crafting_inv );
```

Comment:
Noted. I've patched it to fix the instance @Coolthulhu noticed testing #17302 but I agree we should move the check higher to prevent future relapse.